src/zenml/integrations/kubeflow/orchestrators/local_deployment_utils.py

```diff
@@ -203,21 +207,53 @@ def start_kfp_ui_daemon(pid_file_path: str, port: int) -> None:
         f"{port}:80",
     ]
 
-    def _daemon_function() -> None:
-        """Port-forwards the Kubeflow Pipelines UI pod."""
-        subprocess.check_call(command)
-
-    if sys.platform == "win32":
+    if not port_available(port):
+        modified_command = command.copy()
+        modified_command[-1] = "PORT:80"
+        logger.warning(
+            "Unable to port-forward Kubeflow Pipelines UI to local port %d "
+            "because the port is occupied. In order to access the Kubeflow "
+            "Pipelines UI at http://localhost:PORT/, please run '%s' in a "
+            "separate command line shell (replace PORT with a free port of "
+            "your choice).",
+            port,
+            " ".join(modified_command),
+        )
+    elif sys.platform == "win32":
         logger.warning(
-            f"Daemon functionality not supported on Windows. "
-            f"In order to access the Kubeflow Pipelines UI, please run "
-            f"'{' '.join(command)}' in a separate command line shell."
+            "Daemon functionality not supported on Windows. "
+            "In order to access the Kubeflow Pipelines UI at "
+            "http://localhost:%d/, please run '%s' in a separate command "
+            "line shell.",
+            port,
+            " ".join(command),
         )
     else:
         from zenml.utils import daemon
 
+        def _daemon_function() -> None:
+            """Port-forwards the Kubeflow Pipelines UI pod."""
+            subprocess.check_call(command)
+
         daemon.run_as_daemon(
-            _daemon_function,
-            pid_file=pid_file_path,
+            _daemon_function, pid_file=pid_file_path, log_file=log_file_path
+        )
+        logger.info(
+            "Started Kubeflow Pipelines UI daemon (check the daemon logs at %s "
+            "in case you're not able to view the UI). The Kubeflow Pipelines "
+            "UI should now be accessible at http://localhost:%d/.",
+            log_file_path,
+            port,
         )
-        logger.info("Started Kubeflow Pipelines UI daemon.")
+
+
+def port_available(port: int) -> bool:
```

Comment:
Can we put this in a more accessible place? Perhaps other orchestrators can use it? I think we can also move some other functions outwards too

Reply:
Sure, something like `utils/networking_utils.py`? Which other functions are you referring to? As far as I can see they're all directly related to starting up kubeflow or k3d

Reply:
I think i was thinking of container_entrypoint and docker_utils , that these should be outside of kubeflow/ as well

Reply:
Container entrypoint requires the `kubernetes` package and also contains code that is exclusively for kubeflow, so I'd say that should definitely stay inside the integration. Where would we put the docker file in your opinion? I wouldn't call it an integration as by itself it does not have any value or purpose inside zenml.